XamarinCommunityToolkit/Core/StreamMediaSource.shared.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Core
+{
+	public class StreamMediaSource : MediaSource, IStreamImageSource
+	{
+		readonly object synchandle = new object();
+		CancellationTokenSource cancellationTokenSource;
+
+		TaskCompletionSource<bool> completionSource;
+
+		public static readonly BindableProperty StreamProperty = BindableProperty.Create(nameof(Stream), typeof(Func<CancellationToken, Task<Stream>>), typeof(StreamMediaSource),
+			default(Func<CancellationToken, Task<Stream>>));
+
+		protected CancellationTokenSource CancellationTokenSource
+		{
+			get => cancellationTokenSource;
+			private set
+			{
+				if (cancellationTokenSource == value)
+					return;
+				if (cancellationTokenSource != null)
+				{
+					cancellationTokenSource.Cancel();
+					cancellationTokenSource.Dispose();
+				}
+				cancellationTokenSource = value;
+			}
+		}
+
+		bool IsLoading => cancellationTokenSource != null;
+
+		public virtual Func<CancellationToken, Task<Stream>> Stream
+		{
+			get => (Func<CancellationToken, Task<Stream>>)GetValue(StreamProperty);
+			set => SetValue(StreamProperty, value);
+		}
+
+		protected override void OnPropertyChanged(string propertyName)
+		{
+			if (propertyName == StreamProperty.PropertyName)
+				OnSourceChanged();
+			base.OnPropertyChanged(propertyName);
+		} 
+
+		async Task<Stream> IStreamImageSource.GetStreamAsync(CancellationToken userToken)
```

Comment:
Do we need to implement this interface Explicitly?

Reply:
It would appear so. I'm not convinced that we need it at all, but let's see if we can clean it up when we get the other platforms in here